src/Umbraco.Web.UI.Client/src/common/services/user.service.js

```diff
@@ -185,7 +185,19 @@ angular.module('umbraco.services')
             authenticate: function (login, password) {
 
                 return authResource.performLogin(login, password)
-                    .then(this.setAuthenticationSuccessful);
+                    .then(function(data) {
+
+                        // Check if user has a start node set.
+                        if(data.startContentIds.length === 0 && data.startMediaIds.length === 0){
+                            var errorMsg = "User has no start-nodes";
+                            var result = { errorMsg: errorMsg, user: data, authenticated: false, lastUserId: lastUserId, loginType: "credentials" };
+                            eventsService.emit("app.notAuthenticated", result);
+                            throw result;
```

Comment:
Do we want to 'throw' here? or return `$q.reject(result)` ?

Reply:
Well, I guess we could change the flow and use reject. But for this, I use the Throw to break the promise flow. This way we avoid calling setAuthenticationSuccessful.

But if you feel like I guess it could be remodeled a bit, but this has the least impact on the code.

Reply:
ok cool, just wanted to check if there was a reason, just not sure if there's really a diff between the two as I would suspect throwing within a promise would call the failed promise result anyways. but all good.